lib/config/validation.ts

```diff
@@ -599,6 +599,22 @@ export async function validateConfig(
                   message: `Invalid \`${currentPath}.${key}.${res}\` configuration: value is not a string`,
                 });
               }
+            } else if (key === 'env') {
+              const allowedEnvVars = GlobalConfig.get('allowedEnv');
+              for (const [envVarName, envVarValue] of Object.entries(val)) {
+                if (!allowedEnvVars?.includes(envVarName)) {
+                  warnings.push({
+                    topic: 'Configuration Error',
+                    message: `Invalid enviroment variable name \`${envVarName}\` found in \`${currentPath}\`. Allowed values are ${allowedEnvVars?.join(', ')}.`,
+                  });
+                }
+                if (!is.string(envVarValue)) {
+                  warnings.push({
+                    topic: 'Configuration Error',
+                    message: `Enviroment variable inside \`${currentPath}.${envVarName}\` should be a string.`,
```

Comment:
```suggestion
                    message: `Enviroment variable inside \`${currentPath}.${envVarName}\` must be a string.`,
```

It sounds like we mean `must` here instead of `should`?

Should leaves room for the user to ignore the message, while must indicates the user should go fix something. :smile: